shared/teams/common/use-autocompleter.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import * as Container from '../../util/container'
+
+const useAutocompleter = (items: Array<string>, onSelect: (value: string) => void, filter: string) => {
+  const [selected, setSelected] = React.useState(0)
+  const filterLCase = filter.trim().toLowerCase()
+  const prevFilterLCase = Container.usePrevious(filterLCase)
+  React.useEffect(() => {
+    if (prevFilterLCase !== filterLCase) {
+      setSelected(0)
+    }
+  }, [setSelected, prevFilterLCase, filterLCase])
+  let itemsFiltered = filter ? items.filter(item => item.toLowerCase().includes(filterLCase)) : items
+  itemsFiltered = itemsFiltered.slice(0, 5)
+
+  const {popup, popupAnchor, setShowingPopup, showingPopup} = Kb.usePopup(
+    getAttachmentRef => (
+      <Kb.Overlay
+        attachTo={getAttachmentRef}
+        visible={showingPopup}
+        onHidden={() => setShowingPopup(false)}
+        matchDimension={true}
+        position="top center"
+        positionFallbacks={['bottom center']}
+      >
+        {itemsFiltered.map((item, idx) => (
+          <Kb.ClickableBox
+            key={item}
+            onMouseDown={() => onSelect(item)}
+            onMouseOver={() => setSelected(idx)}
+            style={styles.optionOuter}
+          >
+            <Kb.Box2
+              direction="horizontal"
+              fullWidth={true}
+              style={Styles.collapseStyles([styles.option, selected === idx && styles.optionSelected])}
+            >
+              <Kb.Text type="BodySemibold" lineClamp={1}>
+                {item}
+              </Kb.Text>
+            </Kb.Box2>
+          </Kb.ClickableBox>
+        ))}
+      </Kb.Overlay>
+    ),
+    filterLCase + selected
```

Comment:
might be cool if this were an array and worked like `deps` does for other hooks.